Admin/Admin.php

```diff
@@ -1742,11 +1742,11 @@ public function setSubject($subject)
     public function getSubject()
     {
         if ($this->subject === null && $this->request) {
-            $id = $this->request->get($this->getIdParameter());
-            if (!preg_match('#^[0-9A-Fa-f]+$#', $id)) {
+            if (!$this->subject = $this->getModelManager()->find(
```

Comment:
This is not very optimal if there is no id parameter in the request. Some check for empty id should be before calling the `find` method.